lib/json_api_resource/resource.rb

```diff
@@ -72,6 +76,9 @@ def method_missing(method, *args, &block)
       else
         super
       end
+
+    rescue JsonApiClient::Errors::ServerError => e
+      pretty_error e
```

Comment:
save, update_attributes, and the instance-level method_missing should all catch ServerError and return status codes like find and where do, instead of the exception coming through.  Then we can use respondable classes to act on this from metadata instead of catching this exception everywhere in applications.